### DISCUSSION TITLE
How to collect thin vs thick provisioned volume information?

### DISCUSSION THREAD

**Discussion by jmg011** [NetApp/harvest#689] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Thin/Thick provisioned volume information.

**Describe alternatives you've considered**
Custom Monitor

**Additional context**
N/A

**Comment by ruanruijuan** [comment]
I checked with an engineer in WAFL volume team and her answer is:
volume-get-iter.volume-clone-info.space-reserve
Space Guarantee Style Attributes: optional-for-create, non-modifiable Possible values:
    "none" ,
    "volume" 
'none' implies thin-provisioned and 'volume' implies thick-provisioned.

(The rest equivalent is guarantee.type. This note is for the Harvest team.)

**Comment by jmg011** [comment]
@ruanruijuan Thank you. Should I enable this in the default template or do you recommend that this should be a custom template for the volume?

**Comment by jmg011** [comment]
@ruanruijuan Hi, I could enable it for c-mode with space attributes, but I could not find the way to collect this data for 7-mode. Can you help get more information on this for 7-mode?

**Comment by ruanruijuan** [comment]
@jmg011 can you please check https://www.netapp.com/pdf.html?item=/media/19678-tr-3965.pdf which I googled?

**Comment by ruanruijuan** [comment, marked as answer]
> 
There is space-reserve attribute in the 7-mode zapi as well and it's under "volume-list-info.volume-info".

**Comment by jmg011** [comment]
Got it working. Thank you @ruanruijuan.
